src/ucp/core/ucp_ep.c

```diff
@@ -2665,6 +2665,30 @@ void ucp_ep_get_tl_bitmap(ucp_ep_h ep, ucp_tl_bitmap_t *tl_bitmap)
     }
 }
 
+const char* ucp_ep_get_lane_info_str(ucp_ep_h ucp_ep, ucp_lane_index_t lane,
+                                     ucs_string_buffer_t *lane_info_strb)
+{
+    ucp_rsc_index_t rsc_index;
+    uct_tl_resource_desc_t *tl_rsc;
+
+    if (lane != UCP_NULL_LANE) {
+        if (lane == ucp_ep_get_cm_lane(ucp_ep)) {
+            ucs_string_buffer_appendf(lane_info_strb, "CM lane");
+        } else {
+            rsc_index = ucp_ep_get_rsc_index(ucp_ep, lane);
+            tl_rsc    = &ucp_ep->worker->context->tl_rscs[rsc_index].tl_rsc;
+
+            ucs_string_buffer_appendf(lane_info_strb,
+                                      UCT_TL_RESOURCE_DESC_FMT,
+                                      UCT_TL_RESOURCE_DESC_ARG(tl_rsc));
+        }
+    } else {
+        ucs_string_buffer_appendf(lane_info_strb, "wireup lane");
+    }
+
+    return ucs_string_buffer_cstr(lane_info_strb);
```

Comment:
IMO better to do this outside of the call, since the pointer has the same scope as the buffer, can return pointer to the same buffer to do like this:
```
ucs_string_buffer_cstr(ucp_ep_get_lane_info_strb(ucp_ep, lane, &lane_info_strb))
```

Reply:
the scope of the string buffer is outside of the `ucp_ep_get_lane_info_strb()` scope, so, it is safe to do like this

Reply:
I disagree, it's safe according your usage, but what if someone returns `char*` from the caller somewhere else. IMO it's unsafe to split main object and its internally incapsulated data.

Reply:
fixed